tests/PHPStan/Analyser/data/mixed-to-number.php

```diff
@@ -0,0 +1,37 @@
+<?php declare(strict_types = 1);
+
+namespace MixedToNumber;
+
+use function PHPStan\Testing\assertType;
+
+class HelloWorld
+{
+	public function sayBar($mixed): void
+	{
+		if (!is_float($mixed)) {
+			assertType('mixed~float', $mixed);
+			$mixed--;
+			assertType('int', $mixed);
+		}
+	}
+
+	public function sayFoo($mixed): void
+	{
+		if (!is_float($mixed)) {
+			assertType('mixed~float', $mixed);
+			$mixed++;
+			assertType('int', $mixed);
+		}
+	}
+
+	public function sayFooBar($mixed): void
+	{
+		if (!is_float($mixed)) {
+			if (!is_int($mixed)) {
+				assertType('mixed~float|int', $mixed);
+				$mixed++;
+				assertType('*NEVER*', $mixed);
```

Comment:
Could still be a string though? https://3v4l.org/D4BHP

Reply:
yeah,.. I think the src/ changes actually makes sense .. but my test is to synthetic and your example shows that we cannot return a `Never` type for `toNumber`